### ISSUE TITLE
How to actually pass data and what shape should it be?

### ISSUE THREAD

**Issue by pawelgnatowski** [micha-lmxt/svelte-window#11]
Hi,
i really like your implementation - looks like performance is great - just curious how to pass actual data - would  you be so kind and provide an example?
I could overwrite some of your implementation, but feels weird and that i just do not understand it fully.

Thanks for any feedback here.
Specifically, Fixed or VariableSizeGrid

**Comment by micha-lmxt** [comment]
I've put an example for the VariableSizeGrid into the readme. I hope it helps. 

There is no rule, how your data should look like. Only thing that is important is that you can find the correct cell via row and column indexes.

I also have more complex use cases in my blog:
https://gradientdescent.de/merging-cells/

**Comment by pawelgnatowski** [comment]
I may have not been clear enough. Let me rephrase:
Lets say i have this data:
```
const renderMeAsFixedOrVariableGrid = [
//first row
["John","Doe","20"],
//second row
["Amber","Gold","300"],
//1M row 
...
]
```
How do i pass this data so it is rendered by the Grid?
```
<Grid
	columnCount={3}
	columnWidth={(index) => columnWidths[index]}
	height={600}
	rowCount={1000_000}
	rowHeight={(index) => rowHeights[index]}
	width={1200}
	overscanRowCount={0}
	overscanColumnCount={0}
    
	let:items
>
	{#each dt as it (it.key)}
		<div style={sty(it.style) + 'background-color:red'}>
			{it.data}
		</div>
	{/each}
</Grid>
```

based on the analysis i have done so far i can pass itemData to grid => single string or a number, not sure how or why I would only pass a scalar instead of the multitude of data I want to render.
Obviously i am missing something here.
let directive is supposed to pass items from parent to grid -> but if i pass it:
`let:items ={renderMeAsFixedOrVariableGrid }`
(as i understand from svelte docs :
https://svelte.dev/tutorial/slot-props)
it is overwritten by assignment inside the Grid component
```
    let request_resetIsScrollingDebounced = false,
        request_getItemStyleCache = false,
        // render state
        items = [],
```
Tbh i am pretty new to Svelte and have zero React experience. Would appreciate your help here.

**Comment by micha-lmxt** [comment]
Np. You don't need to pass the data to the grid component. You can retain the data from your 2d array via the indexes:
```svelte
<Grid
	columnCount={3}
	columnWidth={(index) => columnWidths[index]}
	height={600}
	rowCount={1000000}
	rowHeight={(index) => rowHeights[index]}
	width={1200}
    
	let:items
>
	{#each items as it (it.key)}
		<div style={sty(it.style) + 'background-color:red'}>
	{renderMeAsFixedOrVariableGrid[it.rowIndex][it.columnIndex]}
		</div>
	{/each}
</Grid>
```

The 'let:items' directive is the Svelte way to pass props to a slot.

**Comment by pawelgnatowski** [comment]
that did the trick , now i get it. Thanks!
